pkg/porter/mixins.go

```diff
@@ -62,6 +63,41 @@ func (p *Porter) ListMixins() ([]mixin.Metadata, error) {
 	return mixins, nil
 }
 
+// SearchMixins searches the internal remote mixins list according to the provided options
+func (p *Porter) SearchMixins(opts mixin.SearchOptions) error {
+	box := packr.New("get.porter.sh/porter/pkg/mixin/remote-mixins", "../mixin/remote-mixins")
+	mixinSearcher := mixin.NewSearcher(box)
+
+	remoteMixins, err := mixinSearcher.Search(opts)
+	if err != nil {
+		return err
+	}
+
+	if len(remoteMixins) == 0 && opts.Name != "" {
```

Comment:
I would move this logic into mixinSearcher.Search and return it as an error from Search. A couple reasons why:

1. When someone else uses this code, i.e. the tests or someone else in porter or a third party developer, they will get this logic for free. Right now only the CLI gets it.
2. By returning it as an error, it will be automatically handled by lines 72-74.
3. When someone searches for a mixin and nothing comes back they will get a non-zero exit code which will be helpful when they are scripting.

Reply:
Good call; updated.